sgxs-tools/src/sgx_detect/tests/mod.rs

```diff
@@ -784,6 +785,151 @@ impl Dependency<RunEnclaveProdWl> for RunEnclaveProd {
     const CONTROL_VISIBILITY: bool = true;
 }
 
+// Data Shield requirements
+#[derive(Debug, Default, DebugSupport, Update)]
+struct DataShield {
+    enclave_os: Status,
+    enclave_manager: Status,
+    build_type: BuildType,
+}
+
+impl Print for DataShield {
+    fn supported(&self) -> Status {
+        self.enclave_os & self.enclave_manager
+    }
+}
+
+// EnclaveOS requirements
+#[derive(Debug, Default, DebugSupport, Update)]
+struct EnclaveOS {
+    graphene: Status,
+    perm_daemon: Status,
+}
+
+impl Print for EnclaveOS {
+    fn supported(&self) -> Status {
+        self.graphene & self.perm_daemon
+    }
+}
+
+#[derive(Clone, Debug, Default, DebugSupport)]
+struct EnvType {
+    build_type: BuildType,
+}
+
+impl Name for EnvType {
+    fn name(&self) -> &'static str {
+        "EnvType for sgx-detect"
+    }
+}
+
+impl Update for EnvType {
+    fn update(&mut self, support: &SgxSupport) {
+        self.build_type = support.build_type;
+    }
+}
+
+impl Print for EnvType {
+    fn supported(&self) -> Status {
+        if self.build_type == BuildType::Generic {
+            return Status::Fatal
+        }
+        Status::Supported
+    }
+}
+
+#[dependency]
+impl Dependency<EnvType> for DataShield {
+    const CONTROL_VISIBILITY: bool = true;
+    fn update_dependency(&mut self, _dependency: &EnvType, support: &SgxSupport) {
```

Comment:
You don't need to define this, since you don't do anything in this function anyway.